Move me to bottom nav

ravishanker opened the pull request:
Moves Me (avatar) to Bottom Navigation as a fourth tab after Notifications on JP app, and in menu under manage on WP app
- Updates MySite Info layout
- Updates Me screen layout

Fixes # pcdRpT-2HW-p2#comment-4875, and this discussion p1686812471360099-slack-C04SFL0RP51

**Jetpack app**
| Me at the Bottom | Me Screen | Dark Mode
|--------|--------|--------|
|![Screenshot_20230618_160803](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/d4296796-00f6-4e29-95de-ee74962c147f)|![Screenshot_20230618_160742](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/33cb9041-7f93-45df-ad1c-e971a66d526e)|![Screenshot_20230618_162015](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/95f20821-e048-449a-9164-e239781ee085)|

**WordPress app**
| Me in menu | Me |
|--------|--------|
|![Screenshot_20230620_171528](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/8d5c6c20-bc31-47df-9762-1da9cc18bfba) | ![Screenshot_20230620_171538](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/87cd83f9-aef5-4437-865d-e63351a16ba3)|


To test:

- Launch Jetpack app, and login
- Verify that Me/Avatar doesn't appear at top right hand corner anymore 
- Verify that Me is in the bottom navigation, next to Notifications
- Tap on Me tab
- Verify that it goes to Me Screen as before

- Launch WordPress app, and login
- Verify that Me/Avatar doesn't appear at top right hand corner anymore 
- Verify that Me is in the menu, under manage category
- Tap on Me row
- Verify that it goes to Me Screen as before

#### To-do for later in a separate PR

- [ ] Replace icons with new ones on Me Screen
- [ ] Add change photo cta on Me -> My Profile screen if required


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

ravishanker committed:
Remove Me avatar on MySite

ravishanker committed:
Add Me to bottom navigation

ravishanker committed:
Remove AppBar from MeFragment

osullivanchris reviewed (approved):
Sharing per slack the reasons I think this is a good move:

1. the iA is more 'correct'. Today, 'Me' is inside My Site, which is the opposite of the actual parent/child structure
2. its more consistent with WordPress.com where the top bar has My Sites, Reader, Me, Notifications
3. When we merged Me into My Site years ago, My Site was just a menu. Now My Site is more complicated and we have several rows of navigations/information at the top
4. We're using a lot of vertical space for only one icon 
5. Removing that vertical space aligns the header information on My Site, Reader, Notifications more consistently

Here are a couple of improvements I think we could make. Wouldn't have to do all immediately and some are separate issues. The padding bug is a must fix of course. Then, I think changing the layout of the profile photo and name on 'Me' to match the My Site header would be a big improvement, if I was to pick one change.

<img width="2290" alt="Me-Tab-Tweaks" src="https://github.com/wordpress-mobile/WordPress-Android/assets/50576199/82fc4f17-e918-4466-bde7-61487b40b10d">

ravishanker committed:
Move Me avatar to Bottom Nav

Moved Me avatar to Bottom Nav

ravishanker committed:
Update MySite Info header card

- Reduce blavatar size from 64 to 40
- Reduce chevron size from 24 to 16

ravishanker committed:
Update Me screen

- Update to new header layout similar to My Site
- Removed CHANGE PHOTO cta
- Updated paddings and margins
- clean up

ravishanker committed:
Remove too tip on FAB

ravishanker committed:
Update dimens.xml

Adjust app_bar_with_site_info_tabs_height from 200 dp to 136dp since Me is removed now

ravishanker committed:
Merge branch 'trunk' into Move-Me-to-Bottom-Nav

ravishanker committed:
Fix detekt reported issues

ravishanker committed:
Update padding

- increase the Site icon on My Site to 56dp
- Increase Me icon size to 56dp, and align row icons to Center of Me icon on Me screen

ravishanker commented:
👋 @ovitrif - I've added you as a reviewer since you're also doing some work in this area. 🙇

ravishanker commented:
👋 @pachlava - I've added you as this might fail instrumented tests, any other automation stuff. 🙇

ravishanker committed:
Update RELEASE-NOTES.txt

AjeshRPai commented:
Hey @ravishanker 👋🏼 

Awesome job on moving me to the bottom nav 🙌🏼 . Sorry for jumping in, but I was doing some changes in the dashboard and noticed this PR. 

FYI - On **jetpack phase removal four**, **new users phase** and **self hosted users phase**, we are removing this bottom nav entirely to prevent the access of reader and notifications. you might need to make some changes to show the bottom nav without a reader and notifications but only with me page and my site bottom nav in the above phases. 

⚠️  🐛  I tested this PR and enabled the `jp_removal_four` flag, and the bottom nav is completely gone. So a user can't access the me page at all. Screenshot below. 

<img src="https://github.com/wordpress-mobile/wordpress-android/assets/17463767/c24d40a6-1b61-487f-9421-6ff406ef12c2" width="300" height="660">

I also noticed the padding issue, but I am unsure if that's a WIP. If thats fixed, let me know so I can double-check. 

also, There is an issue int the enabling of phase 4 right now which we are trying to fix with https://github.com/wordpress-mobile/WordPress-Android/pull/18612, 


TO reproduce the issue of the me tab not shown, you can try any of the below scenarios
1. ~Disable `jp_removal_static_posters_phase` and enable `jp_removal_four`~
2. ~IF you are testing this scenario after this [PR](https://github.com/wordpress-mobile/wordpress-android/pull/18612) is merged, then just~ enable `jp_removal_four`
3. enable `jp_removal_new_users` - this phase is rolled out, in this phase notifications and reader is not shown 
4. enable `jp_removal_self_hosted_users`- this phase is rolled out as well, in this phase notifications and reader is not shown as well 

Let me know if you need any help with the changes that need to be done or want more information on this.

osullivanchris commented:
@AjeshRPai thanks for taking the time to look and highlighting the changes on the WordPress side. That wasn't front in my mind. I'm going to take a look at it. But I think the 'default' solution would be having a tab bar with 2 tabs as you described.

Another approach would be to keep the old layout with 'me' in the My Site navigation bar only for WordPress, but that seems like more differences between the two apps to support and maintain than a tab bar with items which turn on and off...

ravishanker commented:
> Let me know if you need any help with the changes that need to be done or want more information on this.

@AjeshRPai - Thanks for noticing, and flagging this early 🙇   I'll need to make some changes then in the light phase four.  I'll add you as a reviewer as well to make sure it all works before and after.  

This can be done in two ways
- Bottom nav with My Site, and Me
- Me under Manage in the menu (ignore profile pic, and padding for now)

| Bottom Nav | Manage |
|--------|--------|
| ![Screenshot_20230620_142832](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/014c5209-74dd-447e-96d3-096d28ee9812)| ![Screenshot_20230620_141159](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/74a7b36f-b389-42ee-99d6-6834a5f1aabf)|

ravishanker committed:
Merge branch 'trunk' into Move-Me-to-Bottom-Nav

ravishanker committed:
Move Me to menu for WP

Me is going under menu on WP,
since Bottom Navigation is going away

ravishanker committed:
Update MePage.kt

Fix test

ovitrif commented:
> 👋 @ovitrif - I've added you as a reviewer since you're also doing some work in this area. 🙇

This is a no-brainer for me, I like it, though as @AjeshRPai shared, there is some overlap with ongoing work to improve the daily development workflow, by optimising and making debug settings easily accessible regardless of logged in/out state.

Then there's also the phase 4, which Chris is aware of.

Thus carry on from my side 🙏🏻 , I'm not needed here since I ended up focusing on crashes and didn't do any Compose refactoring 😅 .

AjeshRPai commented on WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:
```diff
@@ -663,14 +660,6 @@ private void initViewModel() {
-            } else {
-                mFabTooltip.hide();
-            }
-
```
👏🏼   thanks for removing this tooltip, ICYMI you can close [this issue](https://github.com/wordpress-mobile/wordpress-android/issues/18249) once this PR is merged 
cc: @thomashorta

AjeshRPai reviewed (changes requested):
Hey @ravishanker 

I tested the following scenarios 
- migration from static posters phase to phase 4| self hosted users phase | new users phase 🟢 
    -  Wordpress app 
       -  Bottom nav is not shown on phase 4 and me item is shown in menu list 🟢 
       - if the user is on me fragment on static posters phase and the app goes to background, when the phase 4 is switched on user is navigated to my site dashboard without bottom nav. 🟢 
   -  Jetpack app 
       - no ui changes when user migrates 🟢 
- if me tab was the last opened tab, when a user opens the app again, me tab is shown 🟢 

Minor issues 
- In wordpress app (static posters phase) me option is show on the menu items as well on bottom nav - as per my understanding it should be shown either on menu items or on bottom nav, correct me if am wrong 
- in both wp/jp app people icon is a dark circle/white circle 

|  light mode | dark mode | 
|---|---|
| ![screenshot_my_site_wordpress__](https://github.com/wordpress-mobile/wordpress-android/assets/17463767/52244ed9-916e-4abe-96aa-6688d202863c)  | ![screenshot_my_site_wordpress_darkmode__](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/c1698981-196a-4b14-b38c-c88362905c5c)  | 

Other than these minor issues, all looks good to me 👍🏼 

cc: @zwarm 
1. for letting you about this change on dashboard, in case if you want to test for any regression issues 
2. also for letting you know that I have tested to make sure that there are no regression issues that we fixed for phase 4.  mainly 👇🏼 
   1. https://github.com/wordpress-mobile/WordPress-Android/issues/17811 
   2. https://github.com/wordpress-mobile/WordPress-Android/pull/17828

tiagomar commented:
Hey @ravishanker, it seems like we could save some vertical space in Landscape mode.

<img width="777" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/42008628/f60a422e-5c9b-4304-8326-dd32cda276d8">

tiagomar commented:
🐛  When selecting `Me` tab, TalkBack reads the `My Site` description.

https://github.com/wordpress-mobile/WordPress-Android/assets/42008628/49453555-cc28-4638-91ad-fee698840832

ravishanker committed:
Fix Me item icon for People item

ravishanker committed:
Fix Me item showing up in menu

Fix Me item showing on WP in both menu and on bottom nav during static posters phase

ravishanker commented:
> Minor issues
> 
> * In wordpress app (static posters phase) me option is show on the menu items as well on bottom nav - as per my understanding it should be shown either on menu items or on bottom nav, correct me if am wrong
> * in both wp/jp app people icon is a dark circle/white circle
> 
👋 @AjeshRPai - Thanks for testing for various phases.  Fixed the those minor issues.  Please review again. 🙇

ravishanker commented:
> Hey @ravishanker, it seems like we could save some vertical space in Landscape mode.

👋 @tiagomar - This is happening on JP app only in landscape mode right?  It is proving difficult fix this due to various feature flags and differences between WP/JP etc.  Need to handle separately perhaps afterwards

ravishanker commented:
> 🐛 When selecting `Me` tab, TalkBack reads the `My Site` description.
> 
>  meTab.mp4

Me tab is reading the content description correctly.  When on Me screen, it is reading the MySite description.  Still investigating where that is coming from!

ravishanker commented:
👋 @pachlava - If you could please take a look at the failing instrumentation tests, if you're back from AFK . 🙇

ravishanker committed:
Fix test

pachlava commented:
I'm taking a look @ravishanker 👋

pachlava committed:
Updated the flow to get back from `Me` screen.

pachlava commented:
@ravishanker 👋 

A 0db191214b702bcccc7a4ddfbf001ba7ed303b71 of yours updates the flow to get _to_ the `Me` screen 👍, however, there remained one thing that also needed an update: a way to go _back_ from `Me` screen.

Previously, this was done by pressing the back button, since the navbar was not available for `Me` screen, and it resulted in `My site` screen being shown. Now, however, pressing back results in the app being minimized. So the back button press had to be replaced with tapping `My Sites`, which I did in 744b3fe874b6b81a5974e72a3fa7b2a5d79ed657. All the tests passed locally for me, 🤞 for the 🟢 Buildkite.

Apparently, the only place where going back from `Me` screen was needed, was after running `e2eLoginWithSelfHostedAccount`. It can be seen from the logs that all the UI tests were failing after this test - since to "log out" from the self-hosted account, we need to open `Me` screen, see that it's a self-hosted account, and then go back to `My sites` and delete a site.

tiagomar commented:
In WordPress app, as long as `Me` is a tab, we probably want to remove the back arrow on top left corner? It will be needed when `Me` is moved to `Menu` though.

<img width="300" src="https://github.com/wordpress-mobile/WordPress-Android/assets/42008628/f48113c0-5542-4d96-a4ec-a2a77e9dce2b">

AjeshRPai committed:
* fix: back button shown on me fragment in wordpress app
Co-authored-by: Ravi <ravi.program@gmail.com>

fixes the back button shown on me fragment in wordpress app in static
 posters phase

AjeshRPai committed:
Merge branch 'trunk' into Move-Me-to-Bottom-Nav

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.kt

AjeshRPai commented:
Hey @ravishanker 

The people icon and the me option in menu is fixed, but I found the same bug as @tiagomar reported here. the back button is visible on wordpress app in static posters phase. In an effort to merge this change before the code freeze on monday and also because your are AFK, I went ahead and fixed it with the commit [60fb503](https://github.com/wordpress-mobile/WordPress-Android/pull/18606/commits/60fb50372a42f39edd5e2b06d3fbf10bd40fe5f8)

@zwarm am adding you as a reviewer, once you have reviewed the changes i have done you can go ahead and merge it. 👍🏼 

I have tested the jetpack app and wordpress app for all other scenarios, everything looks good to me. 🟢 

@tiagomar if you can review it once again, it would be awesome 🙌🏼

AjeshRPai committed:
* suppress: lint error

zwarm reviewed (approved):
@AjeshRPai - I've verified the following
- The back button does not show in JP/WP when "me" is a tab
- The back button shows in WP when "me" is in the menu
👍 

@ravishanker @osullivanchris - love this change. 🙇‍♀️

tiagomar commented:
> @tiagomar if you can review it once again, it would be awesome 🙌🏼

It was all good, @AjeshRPai! 👍 

> Me tab is reading the content description correctly. When on Me screen, it is reading the MySite description. Still investigating where that is coming from!

Yep, I've filed a ticket (#18698 ) for better tracking.
